peer/peer.go

```diff
@@ -983,6 +999,13 @@ func (p *Peer) readMessage() (wire.Message, []byte, error) {
 		return nil, nil, err
 	}
 
+	// Calculate and store the message hash of any mixing message
+	// immediately after deserializing it.
+	switch msg := msg.(type) {
```

Comment:
Not a big deal, but I'd prefer a type assertion as opposed to a single case switch since the latter sets off linters:

```Go
if msg, ok := msg.(hashable); ok {
	msg.WriteHash(p.blake256Hasher)
}
```